great_expectations/datasource/fluent/sources.py

```diff
@@ -542,15 +541,13 @@ def add_or_update_datasource(
             )
 
             # preserve any pre-existing id for usage with cloud
-            id_: uuid.UUID | None = getattr(
-                self._data_context.datasources.get(datasource_name), "id", None
-            )
+            id_: uuid.UUID | None = getattr(self.all().get(datasource_name), "id", None)
```

Comment:
Doesn't this mean that we are now requesting every single datasource anytime we need to update a single field on a single datasource?

Reply:
The behavior should be identical. Previously we had on abstract data context:
```
def datasources(self) -> DatasourceDict:
    return self._datasources
```

Now we have on `_SourceFactories`:
```
    def all(self) -> DatasourceDict:
        return self._data_context._datasources
```

The actual requests are managed internal to `context._datasources`.